src/tic_tac_toe/cli_ui.clj

```diff
@@ -0,0 +1,118 @@
+(ns tic-tac-toe.cli-ui
+  (:require [tic-tac-toe.messages :refer [ask-for-tile-choice
+                                          ask-replay
+                                          draw-message
+                                          goodbye
+                                          invalid-mode
+                                          picked-tile-message
+                                          not-integer-message
+                                          not-y-or-n
+                                          welcome-message
+                                          winner-message]]))
+
+(defn clear-screen []
+  (print (str (char 27) "[2J")))
+
+(defn delay-in-secs 
+  [secs]
+  (Thread/sleep (* secs 1000)))
+
+(defn format-board-cli
+  [board-values]
+  (let [board-size (int (Math/sqrt (count board-values)))]
+    (->> board-values
+         (map (fn [x] (str x " ")))
+         (partition board-size)
+         (interpose "\n")
+         (apply concat)
+         clojure.string/join)))
+
+(defn send-message
+  [message]
+  (println message))
+
+(defn get-string-from-user []
+  (read-line))
+
+(defn attempt-get-number
+  []
```

Comment:
It's minor, but there's inconsistencies regarding which lines you put your arguments on. Most empty argument vectors seem to be on the same line as the function name except here, but in any case it looks odd to allow both styles, so it might be worth spending a hot minute or two relocating your `[]` arguments to the line below (or the non-empty ones to the line above).

Reply:
okay thanks, I thought - why waste a line with empty brackets and having arguments, where they exist, below looked more readable to me - however I'll just move all parameters to the same line as the function name so that at least there's no confusion about which way I'm doing it